File: Client_Root/Client/Assets/Scripts/Network/Messages/ToRoom/GameEventDashToR.cs
using UnityEngine;
using FlatBuffers;

public class GameEventDashToR : IMessage
{
    public const ushort MESSAGE_ID = MessageID.GameEventDashToR_ID;

    public int m_nPlayerIndex;

    public override ushort GetID()
    {
        return MESSAGE_ID;
    }

    public override IMessage Clone()
    {
        return null; 
    }

    public override byte[] Serialize()
    {
        FBS.GameEventDashToR.StartGameEventDashToR(m_Builder);
        FBS.GameEventDashToR.AddPlayerIndex(m_Builder, m_nPlayerIndex);
        var data = FBS.GameEventDashToR.EndGameEventDashToR(m_Builder);

        m_Builder.Finish(data.Value);

        return m_Builder.SizedByteArray();
    }

    public override bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = FBS.GameEventDashToR.GetRootAsGameEventDashToR(buf);

        m_nPlayerIndex = data.PlayerIndex;

        return true;
    }
}

File: Server_Root/Server/RoomServer/Messages/ToRoom/GameEventDashToR.cpp
#include "stdafx.h"
#include "GameEventDashToR.h"
#include "../../../FBSFiles/GameEventDashToR_generated.h"

GameEventDashToR::GameEventDashToR()
{
}

GameEventDashToR::~GameEventDashToR()
{
}

unsigned short GameEventDashToR::GetID()
{
	return MESSAGE_ID;
}

IMessage* GameEventDashToR::Clone()
{
	GameEventDashToR* pClone = new GameEventDashToR();

	pClone->m_nPlayerIndex = m_nPlayerIndex;

	return pClone;
}

const char* GameEventDashToR::Serialize(int* pLength)
{
	FBS::GameEventDashToRBuilder data_builder(m_Builder);
	data_builder.add_PlayerIndex(m_nPlayerIndex);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool GameEventDashToR::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<FBS::GameEventDashToR>((const void*)pChar);

	m_nPlayerIndex = data->PlayerIndex();

	return true;
}